lib/cache/ets.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Provider.Cache.ETS do
+  @moduledoc """
+  An ets based cache implementation
+  """
+  @behaviour Provider.Cache
+
+  use GenServer
+
+  @spec start_link(Keyword.t()) :: GenServer.server()
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, :ok, name: __MODULE__)
+  end
+
+  @impl Provider.Cache
+  def set(module, key, value) do
+    GenServer.call(__MODULE__, {:set, module, key, value})
+  end
+
+  @impl Provider.Cache
+  def get(module, key) do
+    case :ets.lookup(__MODULE__, {module, key}) do
+      [{{^module, ^key}, value}] -> {:ok, value}
+      [] -> {:error, :not_found}
+    end
+  end
+
+  @impl GenServer
+  def init(:ok) do
+    state = :ets.new(__MODULE__, [:named_table])
```

Comment:
It's probably worth turning `:read_concurrency` for this table, as it will be mostly used for (possibly frequent) reads by many processes.